validators/cgroup_validator_test.go

```diff
@@ -93,3 +94,73 @@ func TestValidateCgroupSubsystem(t *testing.T) {
 		})
 	}
 }
+
+func TestGetUnifiedMountpoint(t *testing.T) {
+	tests := map[string]struct {
+		mountsFileContent string
+		expectedErr       bool
+		expectedPath      string
+	}{
+		"cgroup v2": {
+			mountsFileContent: "cgroup2 /sys/fs/cgroup cgroup2 rw,seclabel,nosuid,nodev,noexec,relatime 0 0",
+			expectedErr:       false,
+			expectedPath:      "/sys/fs/cgroup",
+		},
+		"cgroup v1": {
+			mountsFileContent: "cgroup /sys/fs/cgroup cgroup rw,seclabel,nosuid,nodev,noexec,relatime 0 0",
+			expectedErr:       false,
+			expectedPath:      "/sys/fs/cgroup",
+		},
+		"empty file": {
+			mountsFileContent: "",
+			expectedErr:       true,
+			expectedPath:      "",
+		},
+		"none cgroup mounts": {
+			mountsFileContent: `proc /proc proc rw,nosuid,nodev,noexec,relatime 0 0
+sysfs /sys sysfs rw,seclabel,nosuid,nodev,noexec,relatime 0 0`,
+			expectedErr:  true,
+			expectedPath: "",
+		},
+		"multiple cgroups v1 and v2": {
+			mountsFileContent: `cgroup /sys/fs/cgroup/cpuset cgroup rw,nosuid,nodev,noexec,relatime,cpuset
+cgroup /sys/fs/cgroup/memory cgroup rw,nosuid,nodev,noexec,relatime,memory
+cgroup2 /sys/fs/cgroup/unified cgroup2 rw,seclabel,nosuid,nodev,noexec,relatime`,
+			expectedErr:  false,
+			expectedPath: "/sys/fs/cgroup/unified",
+		},
+		"cgroup v1 only with multiple subsystems": {
+			mountsFileContent: `cgroup /sys/fs/cgroup/cpuset cgroup rw,nosuid,nodev,noexec,relatime,cpuset
+cgroup /sys/fs/cgroup/memory cgroup rw,nosuid,nodev,noexec,relatime,memory`,
+			expectedErr:  false,
+			expectedPath: "/sys/fs/cgroup/cpuset", // First valid cgroup v1 path
+		},
+		"no valid cgroup": {
+			mountsFileContent: "proc /proc proc rw,nosuid,nodev,noexec,relatime 0 0\nsysfs /sys sysfs rw,seclabel,nosuid,nodev,noexec,relatime 0 0",
+			expectedErr:       true,
+			expectedPath:      "",
+		},
+	}
+
+	for desc, test := range tests {
+		t.Run(desc, func(t *testing.T) {
+			tmpFile, err := os.CreateTemp("", "mounts")
+			assert.NoError(t, err, "Unexpected error creating temp file")
+			defer os.Remove(tmpFile.Name())
+
+			_, err = tmpFile.Write([]byte(test.mountsFileContent))
+			assert.NoError(t, err, "Unexpected error writing to temp file")
+			tmpFile.Close()
+
+			path, err := getUnifiedMountpoint(tmpFile.Name())
+
+			if test.expectedErr {
+				assert.Error(t, err, "%q: Expected error but got none", desc)
+			} else {
+				assert.NoError(t, err, "%q: Did not expect error but got one: %s", desc, err)
+			}
+
+			assert.Equal(t, test.expectedPath, path, "%q: Expected cgroup path mismatch", desc)
+		})
```

Comment:
we can remove the parameter `desc`  when doing assertion e.g.
`assert.Equal(t, test.expectedPath, path, "%q: Expected cgroup path mismatch")`

because the subtest `t.Run(desc, func(t *testing.T) {` already tells which test is currently running.

Reply:
Good Catch.